### ISSUE TITLE
Message Bar and link colors do not work well in high contrast mode

### ISSUE THREAD

**Issue by tipantaz** [OfficeDev/office-ui-fabric-react#10493]
Hello team!
I couldn't find a duplicate of this but there are some bugs that seem related.

The bug is the following:
When I have a link in the message bar and specify a color, in high contrast mode black it does not work properly on edge.

It stays the same color while the background becomes white as well.

https://codepen.io/tipantaz/pen/MWgPvjj

Actual behavior:
Link color stays the same.

Expected behavior:
Expecting to change color according to the background.

Thanks!

**Comment by aneeshack4** [comment]
@tipantaz Thanks for reporting this! I believe the MessageBar styles are missing a HighContrastSelector - I will send a fix out.

**Comment by aneeshack4** [comment]
@tipantaz This is actually because you're not using Fabric's `Link` control - when I changed the anchor to `Link` it is visible:
![image](https://user-images.githubusercontent.com/4161791/65642579-7dd12580-dfa4-11e9-858d-d260826125c1.png)

**Comment by tipantaz** [comment]
Hey @aneeshack4, sorry for taking a while.
I only used the generic <a> because in order to change the color of the link in a message bar I had to use the !important tag and I suspected that that might be the cause of the problem. 

It might be that I am using the styling wrong even though it does work properly when not in the message bar.

If you could take a look at this https://codepen.io/tipantaz/pen/MWgPvjj

And here are some screenshots for reference

![HighContrast](https://user-images.githubusercontent.com/52921554/65865746-1506f600-e374-11e9-991b-e140a4361328.PNG)
![NoContrast](https://user-images.githubusercontent.com/52921554/65865748-1506f600-e374-11e9-9d28-31927f81e8a7.PNG)

What do you think?
Thanks for your help and patience!

**Comment by aneeshack4** [comment]
@tipantaz So it's working properly with the `!important` tag? And if not, have you tried what I suggested before with providing your own high contrast selector in the styling for the `Link` component?

**Comment by tipantaz** [comment]
Hi again @aneeshack4 and thanks for the feedback. 
No it is not working with the !important tag in high contrast because the color is not changing, it stays white. 
I think that me having to use !important in the style is a bug itself, what do you think? Is there any better option? If we solve this then we solve the a11y bug as well.

I included everything I could think of in the codepen above.

I will try using the selectors and come back to you.

**Comment by aneeshack4** [comment]
I don't think there's anything wrong with the usage of `!important` but I personally haven't used it with the Fabric `Link` component so perhaps owner @khmakoto may have come across this more? The better option is what I suggested - the high contrast selectors; that is typically how a11y styling is addressed. Please try that.

**Comment by khmakoto** [comment]
@tipantaz I think @aneeshack4 is right here. From my understanding this is how styling is applied when dealing with High Contrast Mode:

1. The default style of the tag you're using is applied.
2. The high contrast style of the tag you're using is applied.
3. The user defined styles you specify are applied.

What this means in this case is that, because you're specifying the style of the link as `color: white`, this becomes the color used in both normal and high contrast modes based on the order of operations above.

I believe that the approach @aneeshack4 suggested here of using high contrast selectors will work here. We have them specified for easy use and you can import them as `HighContrastSelector`, `HighContrastSelectorWhite` and `HighContrastSelectorBlack` from our styling library `@uifabric/styling` which is also exported from the main `office-ui-fabric-react` package.

Alternatively, if you want to write the selectors yourself, their values are the following:

```ts
export const HighContrastSelector = '@media screen and (-ms-high-contrast: active)';
export const HighContrastSelectorWhite = '@media screen and (-ms-high-contrast: black-on-white)';
export const HighContrastSelectorBlack = '@media screen and (-ms-high-contrast: white-on-black)';
```
I hope this helps you solve your issue!

**Comment by khmakoto** [comment]
Upon further review, I do think that there might indeed be a CSS specificity problem here. The main issue is that the `MessageBar` has this styling defined in lines `134-143`:

```ts
selectors: {
   '& .ms-Link': {
      color: palette.themeDark,
      fonts.small
   },
   [HighContrastSelector]: {
      background: 'WindowText',
      color: 'Window'
   }
}
```

Which is of a greater specificity than the styling applied directly over `Link` (two classes vs one). @aneeshack4 I don't know how to cleanly fix this but we can bounce off ideas to come up with a good solution.

**Comment by ecraig12345** [comment]
Talked to Aneesha and I'm going to work on this along with the other MessageBar/Link color issues.

**Comment by ecraig12345** [comment]
Since the discussion on this one got a bit long and hard to follow, I'm going to close this one and track under the very similar issue #10880 instead.
